router/batchrouter/asyncdestinationmanager/salesforce-bulk/api_service.go

```diff
@@ -0,0 +1,279 @@
+			Category:   "ServerError",
+		}
+	}
+
+	return &jobResp, nil
+}
+
+func (s *SalesforceAPIService) GetFailedRecords(jobID string) ([]map[string]string, *APIError) {
+	endpoint := fmt.Sprintf("https://default.salesforce.com/services/data/%s/jobs/ingest/%s/failedResults", s.apiVersion, jobID)
+	return s.getCSVRecords(endpoint)
+}
+
+func (s *SalesforceAPIService) GetSuccessfulRecords(jobID string) ([]map[string]string, *APIError) {
+	endpoint := fmt.Sprintf("https://default.salesforce.com/services/data/%s/jobs/ingest/%s/successfulResults", s.apiVersion, jobID)
+	return s.getCSVRecords(endpoint)
+}
+
+func (s *SalesforceAPIService) DeleteJob(jobID string) *APIError {
+	endpoint := fmt.Sprintf("https://default.salesforce.com/services/data/%s/jobs/ingest/%s", s.apiVersion, jobID)
+
+	_, apiErr := s.makeRequest("DELETE", endpoint, bytes.NewReader([]byte{}), "")
+	return apiErr
+}
+
+func (s *SalesforceAPIService) getCSVRecords(endpoint string) ([]map[string]string, *APIError) {
+	respBody, apiErr := s.makeRequest("GET", endpoint, bytes.NewReader([]byte{}), "")
+	if apiErr != nil {
+		return nil, apiErr
+	}
+
+	reader := csv.NewReader(bytes.NewReader(respBody))
+
+	headers, err := reader.Read()
+	if err == io.EOF {
+		return []map[string]string{}, nil
+	}
+	if err != nil {
+		return nil, &APIError{
+			StatusCode: 500,
+			Message:    fmt.Sprintf("reading CSV header: %v", err),
+			Category:   "ServerError",
+		}
+	}
+
+	var records []map[string]string
+	for {
+		row, err := reader.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, &APIError{
+				StatusCode: 500,
+				Message:    fmt.Sprintf("reading CSV row: %v", err),
+				Category:   "ServerError",
+			}
+		}
+
+		record := make(map[string]string)
+		for i, value := range row {
+			if i < len(headers) {
+				record[headers[i]] = value
+			}
+		}
+		records = append(records, record)
+	}
+
+	return records, nil
+}
+
+func (s *SalesforceAPIService) makeRequest(
+	method, endpoint string,
+	body io.Reader,
+	contentType string,
+) ([]byte, *APIError) {
+	respBody, apiError := s.attemptRequest(method, endpoint, body, contentType)
+	if apiError == nil {
+		return respBody, nil
+	}
+
+	return nil, apiError
+}
+
+func (s *SalesforceAPIService) attemptRequest(
+	method, endpoint string,
+	body io.Reader,
+	contentType string,
+) ([]byte, *APIError) {
+	req, err := http.NewRequest(method, endpoint, body)
+	if err != nil {
+		return nil, &APIError{
+			StatusCode: 500,
+			Message:    fmt.Sprintf("creating HTTP request: %v", err),
+			Category:   "ServerError",
+		}
+	}
+
+	if contentType != "" {
+		req.Header.Set("Content-Type", contentType)
+	}
+
+	req = req.WithContext(cntx.CtxWithDestInfo(req.Context(), s.destinationInfo))
+
+	resp, err := s.client.Do(req)
+	if err != nil {
+		return nil, &APIError{
+			StatusCode: 500,
+			Message:    fmt.Sprintf("making HTTP request: %v", err),
+			Category:   "ServerError",
+		}
+	}
+	defer resp.Body.Close()
```

Comment:
let's explicitly ignore error.
```suggestion
	defer func() { _ = resp.Body.Close() }()
```